cclib/parser/gaussianparser.py

```diff
@@ -213,6 +216,229 @@ def extract(self, inputfile, line):
+        # XZ=     0.0000   YZ=     0.0000   ZZ=    25.1622
+        # Eigenvalues:    25.1622    26.1748    33.2220
+        # End of Minotr F.D. properties file   721 does not exist.
+        # End of Minotr F.D. properties file   722 does not exist.
+        # End of Minotr F.D. properties file   788 does not exist.
+        #
+        # **********************************************************************
+        if 'SCF GIAO Magnetic shielding tensor (ppm)' in line:
+            line = inputfile.next()
+            nmr_iso = []
+            nmr_anis = []
+            nmr_eigen = []
+            while '*************************' not in line:
+                if 'Isotropic' in line:
+                    nmr_iso.append(float(line.split()[4]))
+                    nmr_anis.append(float(line.split()[7]))
+                elif 'Eigenvalues' in line:
+                    nmr_eigen.append([float(line.split()[1]), float(line.split()[2]), float(line.split()[3])])
+                line = inputfile.next()
+            self.set_attribute('nmr_iso', nmr_iso)
+            self.set_attribute('nmr_anis', nmr_anis)
+            self.set_attribute('nmr_eigen', nmr_eigen)
+
+        # Extract Wiberg bond orders
+        #  Wiberg bond index matrix in the NAO basis:
+        #
+        #     Atom    1       2       3       4       5       6       7       8       9
+        #     ---- ------  ------  ------  ------  ------  ------  ------  ------  ------
+        # 1.  C  0.0000  1.0207  0.0078  0.0009  0.0002  0.0016  0.0005  0.0095  0.0024
+        # 2.  C  1.0207  0.0000  0.9907  0.0118  0.0063  0.0012  0.0039  0.0115  0.0089
+        # 3.  C  0.0078  0.9907  0.0000  0.9527  0.0217  0.0212  0.0232  1.0469  0.7798
+        # 4.  C  0.0009  0.0118  0.9527  0.0000  0.8700  0.8732  0.8783  0.0121  0.0096
+        # 5.  F  0.0002  0.0063  0.0217  0.8700  0.0000  0.0463  0.0453  0.0038  0.0017
+        # 6.  F  0.0016  0.0012  0.0212  0.8732  0.0463  0.0000  0.0469  0.0079  0.0037
+        # 7.  F  0.0005  0.0039  0.0232  0.8783  0.0453  0.0469  0.0000  0.0021  0.0074
+        # 8.  C  0.0095  0.0115  1.0469  0.0121  0.0038  0.0079  0.0021  0.0000  0.7958
+        # 9.  C  0.0024  0.0089  0.7798  0.0096  0.0017  0.0037  0.0074  0.7958  0.0000
+        if 'Wiberg bond index matrix' in line:
+            WBO_matrix = []
+            while not 'Wiberg bond index,' in line:
+                line = inputfile.next()
+                if 'Atom' in line:
+                    line = inputfile.next()
+                    line = inputfile.next()
+                    WBO_block = []
+                    while len(line) > 1:
+                        WBO_block.append(line.split()[2:])
+                        line = inputfile.next()
+                    WBO_block = numpy.array(WBO_block)
+                    WBO_block = WBO_block.transpose().tolist()
+                    WBO_matrix += WBO_block
+            self.set_attribute('WBO_matrix', WBO_matrix)
```

Comment:
Yes, this is an example:

[WBO.log](https://github.com/cclib/cclib/files/12496500/WBO.log)

Reply:
Great thanks! Suggestion for new naming scheme:

```suggestion
        if 'Wiberg bond index matrix' in line:
            WBO_matrix = []
            while not 'Wiberg bond index,' in line:
                line = inputfile.next()
                if 'Atom' in line:
                    line = inputfile.next()
                    line = inputfile.next()
                    WBO_block = []
                    while len(line) > 1:
                        WBO_block.append(line.split()[2:])
                        line = inputfile.next()
                    WBO_block = numpy.array(WBO_block)
                    WBO_block = WBO_block.transpose().tolist()
                    WBO_matrix += WBO_block
                self.bondorders['wiberg'] = WBO_matrix
```

There are two things I think we need for this section:
1.  Presumably we need to know which atoms are involved in the bonding, which I think is currently missing? Easiest way to do this is to make `bondorders['wiberg']` a dictionary, and each key can be a `set` of the two atom indices, which should take care of duplicates quite nicely
2. Some tests to make sure we're parsing the information correctly. Have you written tests for cclib before?

Reply:
I would *love* to have bondorders['wiberg'] and/or bondorders['mayer'] .. would make bond order perception so much easier.

Reply:
This is a little problematic since we have bondorders as computational methods and now we would have them as attributes. I can rationalize this being fine since one knows if it was parsed if it is an attribute. Wiberg is nontrivial to compute since you need a concept of a localized basis I think?

To keep this moving forward, lets keep this as a dictionary as suggested by @oliver-s-lee